src/main/scala/com/kunyan/userportrait/importdata/extractor/Extractor.scala

```diff
@@ -0,0 +1,983 @@
+package com.kunyan.userportrait.importdata.extractor
+
+import com.kunyan.userportrait.log.PLogger
+import com.kunyan.userportrait.util.StringUtil
+
+/**
+  * Created by C.J.YOU on 2016/4/27.
+  * 解析电信原始用户数据信息的主类
+  */
+object Extractor extends Serializable {
+
+  /**
+    * 解析出用户phone，qq，微博号
+    * @param arr 电信数据按分隔符分隔后的数组
+    * @return 用户信息元组
+    */
+  def extractorUserInfo(arr: Array[String]): (String, String, String) = {
+
+    (phone(arr), qq(arr), weibo(arr))
+
+  }
+
+  /**
+    * 获取maimai用户的id
+    * @param array 电信原始数据的数组形式
+    * @return  （返回uid，ua，cookie）
+    */
+  def maimaiUserId(array: Array[String]): (String,String,String)  = {
+
+    var info = ""
+    val cookie = array(6)
+
+    if (cookie.contains ("koa:sess=")) {
+
+      try {
+
+        val arr = cookie.split ("koa:sess=")
+
+        if (arr.length > 1) {
+
+          val value = arr(1).split(";")(0)
+
+          if (value.nonEmpty) {
+            info = StringUtil.getMaimaiUserId(value)
+          }
+        }
+      } catch {
+
+        case e:Exception => PLogger.warn("maimai userid error")
+
+      }
+    }
+
+    (info,array(2),array(6))
+
+  }
+
+  /**
+    * 通过其他网站提取用户手机号
+    * @param array 电信数据按分隔符分隔后的数组
+    * @return 用户手机号
+    */
+  private def phoneFromOtherWebsite(array: Array[String]): String = {
```

Comment:
怎么抽象


Sent from my iPhone

> On Jun 21, 2016, at 8:10 PM, 杨帅 <notifications@github.com> wrote:
> 
> In src/main/scala/com/kunyan/userportrait/importdata/extractor/Extractor.scala:
> 
> > +
> > +        case e:Exception => PLogger.warn("maimai userid error")
> > +
> > +      }
> > +    }
> > +
> > +    (info,array(2),array(6))
> > +
> > +  }
> > +
> > +  /**
> > +    * 通过其他网站提取用户手机号
> > +    * @param array 电信数据按分隔符分隔后的数组
> > +    * @return 用户手机号
> > +    */
> > +  private def phoneFromOtherWebsite(array: Array[String]): String = {
> 这个方法里面一大堆if else你自己看着不烦吗，不会抽象一下吗
> 
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub, or mute the thread.
>